### PR TITLE
generic: MT7530 DSA driver backports, fixes and enhancements

### PR DESCRIPTION
**Pull request by dangowrt**
Clean up our downstream mess and sync the MT7530 DSA switch driver with upstream Linux.
Import two patches taking care of long-standing problems with STP (and similar protocols) as well as adding the ability to influence and inquire EEE settings via ethtool.

### CODE REVIEW

**Comment by rsalvaterra** [comment]
I'll give this a spin on the R3 and the RT3200, thanks!

File: target/linux/generic/backport-6.1/790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+ 
+-	/* Disable LEDs before reset to prevent the MT7530 sampling a
+-	 * potentially incorrect HT_XTAL_FSEL value.
+-	 */
+-	mt7530_write(priv, MT7530_LED_EN, 0);
+-	usleep_range(1000, 1100);
+-
+ 	/* Reset whole chip through gpio pin or memory-mapped registers for
+ 	 * different type of hardware
+ 	 */
```
Comment (arinc9) [inline target/linux/generic/backport-6.1/790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch:154]: I would just remove this patch and 790-46-v6.9-net-dsa-mt7530-disable-LEDs-before-reset.patch.

Comment (dangowrt) [reply target/linux/generic/backport-6.1/790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch:154]: I like that all patches are now directly generated via `git format-patch` and are literally almost unchanged upstream commits. Removing that patch means also having to modify the follow-up patch which combines the revert with applying the alternative approach you were in favor of.

Comment (arinc9) [reply target/linux/generic/backport-6.1/790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch:154]: The follow-up patch being "790-46-v6.9-net-dsa-mt7530-disable-LEDs-before-reset.patch"? You wouldn't have to modify that but just get rid of it. Currently, I see two patch files for a single patch. 789-STABLE-01-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch and 790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch. Another option should be to remove 789-STABLE-01-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch.

Comment (dangowrt) [reply target/linux/generic/backport-6.1/790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch:154]: Thanks for pointing that out.

I'll kept only the `789-STABLE-01-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch` and removed the redundant `790-46-v6.9-net-dsa-mt7530-disable-LEDs-before-reset.patch` and `790-47-v6.9-net-dsa-mt7530-prevent-possible-incorrect-XTAL-frequ.patch`

**Pull request merged**